superenv: add HOMEBREW_DYNAMIC_LINKER

xu-cheng opened the pull request:
This env is used to set the dynamic linker in superenv. By introducing
it, we can:
* avoid stdenv for building glibc
* allow overwriting the dynamic linker for formulae in
  homebrew/homebrew-portable.

sjackman commented on Library/Homebrew/extend/os/linux/extend/ENV/super.rb:
```diff
@@ -20,4 +20,12 @@ def determine_rpath_paths
   end
+
+  def determine_dynamic_linker_path(formula)
+    if formula && ["glibc", "glibc25"].include?(formula.name)
```
`glibc25` is an anachronism. I intended using it for building portable bottles, but we never did. You can remove it from here.

sjackman replied:
It was a good idea though, that could maybe be requested one day. In particular, `binutils` is installed before `glibc` is installed, so we bottle it on CentOS 5 rather than on our usual Ubuntu 14. See here https://github.com/Linuxbrew/brew/wiki/Build-a-portable-bottle
It'd be great if it were as easy as:
```ruby
depends_on "glibc@2.5" if build.bottle?
```
That may take a fair bit of work though.

xu-cheng replied:
Updates. However, I have a wild idea. Would it be better if we have portable version of binutils and even glibc which can be poured everywhere on the file system?

sjackman replied:
Both `binutils` and `gcc` are relocatable. In Linuxbrew we've manually marked them both as `cellar :any`, even though they're identified automatically as being no-relocatable. `binutil` is additionally portable because we compile it on CentOS 5. The `gcc` bottle depends on `glibc`, so it is relocatable but not portable.

It would be *fantastic* to make `glibc` relocatable. `glibc` itself does not support being moved to a new directory after it's configured and installed. We'd have to write that code. I'm sure it's possible, but not a small undertaking. The autoconf constant `@sysconfdir@` would have to be replaced with a function that determines at run-time where it's located in the file system, so that it can find its configuration files, libraries and stuff.

sjackman reviewed (changes requested):
Looks great. Thanks for this work, Xu.

xu-cheng committed:
superenv: add HOMEBREW_DYNAMIC_LINKER

This env is used to set the dynamic linker in superenv. By introducing
it, we can:
* avoid stdenv for building glibc
* allow overwriting the dynamic linker for formulae in
  homebrew/homebrew-portable.

sjackman commented:
Except for the one minor change requested, I'm happy with this patch.

sjackman commented:
Thanks, Xu!
